stdlib/public/Backtracing/Backtrace.swift

```diff
@@ -0,0 +1,496 @@
+  #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+  private static func withDyldProcessInfo<T>(for task: task_t,
+                                             fn: (OpaquePointer?) throws -> T)
+    rethrows -> T {
+    var kret: kern_return_t = KERN_SUCCESS
+    let dyldInfo = _dyld_process_info_create(task, 0, &kret)
+
+    if kret != KERN_SUCCESS {
+      fatalError("error: cannot create dyld process info")
+    }
+
+    defer {
+      _dyld_process_info_release(dyldInfo)
+    }
+
+    return try fn(dyldInfo)
+  }
+  #endif
+
+  @_spi(Internal)
+  public static func captureImages(for process: Any) -> [Image] {
+    var images: [Image] = []
+
+    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    let task = process as! task_t
+
+    withDyldProcessInfo(for: task){ dyldInfo in
+      _dyld_process_info_for_each_image(dyldInfo) {
+        (machHeaderAddress, uuid, path) in
+
+        if let path = path, let uuid = uuid {
+          let pathString = String(cString: path)
+          let theUUID = Array(UnsafeBufferPointer(start: uuid,
+                                                count: MemoryLayout<uuid_t>.size))
+          let name: String
+          if let slashIndex = pathString.lastIndex(of: "/") {
+            name = String(pathString.suffix(from:
+                                              pathString.index(after:slashIndex)))
+          } else {
+            name = pathString
+          }
+
+          // Find the end of the __TEXT segment
+          var endOfText = machHeaderAddress + 4096
+
+          _dyld_process_info_for_each_segment(dyldInfo, machHeaderAddress){
+            address, size, name in
+
+            if let name = String(validatingUTF8: name!), name == "__TEXT" {
+              endOfText = address + size
+            }
+          }
+
+          images.append(Image(name: name,
+                              path: pathString,
+                              buildID: theUUID,
+                              baseAddress: machHeaderAddress,
+                              endOfText: endOfText))
+        }
+      }
+    }
+    #endif // os(macOS) || os(iOS) || os(watchOS)
+
+    return images.sorted(by: { $0.baseAddress < $1.baseAddress })
+  }
+
+  /// Capture shared cache information.
+  ///
+  /// @returns A `SharedCacheInfo`.
+  public static func captureSharedCacheInfo() -> SharedCacheInfo {
+    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    return captureSharedCacheInfo(for: mach_task_self_)
+    #else
+    return SharedCacheInfo()
+    #endif
+  }
+
+  @_spi(Internal)
+  public static func captureSharedCacheInfo(for t: Any) -> SharedCacheInfo {
+    #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+    let task = t as! task_t
+    return withDyldProcessInfo(for: task){ dyldInfo in
+      var cacheInfo = dyld_process_cache_info()
+      _dyld_process_info_get_cache(dyldInfo, &cacheInfo)
+      let theUUID = withUnsafePointer(to: cacheInfo.cacheUUID){
+        $0.withMemoryRebound(to: UInt8.self,
```

Comment:
You don't need the rebinding here. You can

```
Array(UnsafeRawBufferPointer(start: $0, count: MemoryLayout<uuid_t>.size))
```

Reply:
Makes sense that would work, and saves rebinding memory.